modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -204,8 +209,10 @@ def is_full_axis_mask(index, axis_length):
         row_labels = [row_labels] if is_scalar(row_labels) else row_labels
         col_labels = [col_labels] if is_scalar(col_labels) else col_labels
 
-        if is_full_axis_mask(row_labels, self._length_cache) and is_full_axis_mask(
-            col_labels, self._width_cache
+        if (
+            not order_broken
```

Comment:
The only way to quickly find out if the order is broken is to ask the layer above that stores the indexes.